.github/actions/observe-build-status/action.yml

```diff
@@ -75,30 +81,33 @@ runs:
         fi
-
-    ####################### DETAILED FLAKY JUNIT TESTS #########################
-    - name: Get detailed info on flaky JUnit tests
-      id: get-detailed-flaky-junit-tests
-      if: ${{ inputs.detailed_junit_flaky_tests == 'true' && steps.secrets.outputs.gcloud_sa_key != '' }}
+    ######################## Report test result #############################  
+    - name: Get detailed info on JUnit tests
+      id: get-detailed-junit-tests
+      if: ${{ (inputs.detailed_junit_tests == 'true' || inputs.detailed_junit_flaky_tests == 'true') && steps.secrets.outputs.gcloud_sa_key != '' }}
       shell: bash
-      # To support multi-line string in output we have to work with EOF delimiter
-      # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#example-of-a-multiline-string
       run: |
-        {
-          echo 'result<<EOF'
-          find . -iname 'TEST-*-FLAKY.xml' | xargs -d '\n' --no-run-if-empty -n1 python3 .github/actions/observe-build-status/junit-flaky-to-jsonl.py
-          (find . -iname 'TEST-*.xml' -exec grep -l '</failure>' {} + | grep -v 'FLAKY.xml' | xargs -d '\n' --no-run-if-empty -n1 python3 .github/actions/observe-build-status/junit-failure-to-jsonl.py || true)
-          echo 'EOF'
-        } >> $GITHUB_OUTPUT
+        TEST_RESULTS_FILE=$(mktemp -t junit_test_results.${GITHUB_JOB}_${GITHUB_RUN_ID}_${GITHUB_RUN_ATTEMPT}.XXXXXX.jsonl)
+        echo "test_results_file=${TEST_RESULTS_FILE}" >> $GITHUB_OUTPUT        
+        find . -iname 'TEST-*.xml' -print0 | xargs -0 python3 .ci/scripts/ci/observe-build-status/junit-test-results-to-jsonl.py > "${TEST_RESULTS_FILE}"
+        echo "Generated $(wc -l < "${TEST_RESULTS_FILE}") test results"
 
-    - uses: camunda/infra-global-github-actions/submit-test-status@main
-      if: ${{ inputs.detailed_junit_flaky_tests == 'true' && steps.secrets.outputs.gcloud_sa_key != '' }}
+    - name: Submit test results
+      uses: camunda/infra-global-github-actions/submit-test-status@main
+      if: ${{ (inputs.detailed_junit_tests == 'true' || inputs.detailed_junit_flaky_tests == 'true') && steps.secrets.outputs.gcloud_sa_key != '' }}
       with:
-        test_event_record: "${{ steps.get-detailed-flaky-junit-tests.outputs.result }}"  # TODO: what is the size limit?
+        test_results_file: "${{ steps.get-detailed-junit-tests.outputs.test_results_file }}"
         job_name_override: "${{ inputs.job_name }}"
         gcp_credentials_json: "${{ steps.secrets.outputs.gcloud_sa_key }}"
-    ############################################################################
 
+    - name: Cleanup test results file
+      if: ${{ (inputs.detailed_junit_tests == 'true' || inputs.detailed_junit_flaky_tests == 'true') }}
+      shell: bash
+      run: |
+        if [ -f "${{ steps.get-detailed-junit-tests.outputs.test_results_file }}" ]; then
+          rm -f "${{ steps.get-detailed-junit-tests.outputs.test_results_file }}"
+        fi    
```

Comment:
🔧 Due to runners getting fresh filesystems for each job, it is not necessary to spend code & time on cleanup here - no secret information is inside those files.